core/src/com/unciv/logic/map/TileInfo.kt

```diff
@@ -795,21 +797,30 @@ open class TileInfo {
     
     fun setTileResource(newResource: TileResource, majorDeposit: Boolean = false) {
         resource = newResource.name
-        
+
         if (newResource.resourceType != ResourceType.Strategic) return
-        
+
         for (unique in newResource.getMatchingUniques(UniqueType.ResourceAmountOnTiles)) {
-            if (matchesTerrainFilter(unique.params[0])) {
+            if (matchesFilter(unique.params[0])) {
```

Comment:
?? In map generation you don't have improvements, so why not use the more specific terrainFilter?